controllers/horizon_controller.go

```diff
@@ -347,21 +346,15 @@ func (r *HorizonReconciler) reconcileNormal(ctx context.Context, instance *horiz
 	// run check OpenStack secret - end
 
 	// Create Memcached instance if no shared instance exists.
-	var memcached *memcachedv1.Memcached
-	if instance.Spec.SharedMemcached == "" {
-		memcached, err = r.ensureHorizonMemcached(ctx, helper, instance)
-	} else {
-		memcached, err = r.getSharedMemcached(ctx, helper, instance)
-	}
-
+	memcached, err := r.getHorizonMemcached(ctx, helper, instance)
```

Comment:
We can probably move this into generateHorizonConfigMap and treat memcached similarly to KeystoneAPI ?

Reply:
Yeah, I think we can move it now that it's been simplified. I think I put it in `reconcileNormal()` initially because it needed to create the Memcached instance before anything else could succeed.

Reply:
A problem I noticed while moving this logic is that `generateServiceConfigMaps` now supports only immediate failure and to support retry after some period we have to modify its interface and also add a new status for ServiceConfigReadyCondition... I probably pend that refactoring and distribute this implementation as it is now.